packages/backend-modules/payments/graphql/resolvers/_mutations/createCheckoutSession.ts

```diff
@@ -25,8 +29,7 @@ export = async function createCheckoutSession(
   Auth.ensureUser(ctx.user)
 
   const shop = new Shop(Offers)
-  const entryOffer =
-    (await utils.hasHadMembership(ctx.user.id, ctx.pgdb)) === false
+  const entryOffer = await checkIntroductoryOfferEligieblity(ctx.pgdb, ctx.user)
```

Comment:
nit: this is named `withEntryOffer` or `withIntroductoryOffer` elsewhere, could update for consistency